### DISCUSSION TITLE
How to build and boot an image from scratch?

### DISCUSSION THREAD

**Discussion by Zibri** [actions/virtual-environments#8969] [General questions] [answered]
Starting from an Ubuntu server minimal installation, how can I use the released JSON (internal.ubuntu-22.04.json for example)
to create an image (locally) and boot from it.
Or how can i use the jon to install and configure a minimal image of Ubuntu 22.04 locally?

**Comment by Zibri** [comment]
Also it would be nice to have a step by step to do the same in virtualbox.

In thi way we could setup a few VMs identical to the ones used by github actions to make tests locally.

**Reply by ilia-shipitsin** [reply, marked as answer]
Hello, @Zibri

The only currently supported and maintained option for building a virtual machine using our repository and sources is presented in the [document](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md#manual-image-generation). Unfortunately, we do not support other methods and, accordingly, cannot answer the question asked. I can separately note that porting an existing solution for use with Virtualbox will be quite difficult.

If you have any other questions or suggestions, feel free to share.
